### PR TITLE
Added Entity, Repository, Controller, and Tests for UCSB Organization

### LINKED ISSUE #22: Add `DELETE` endpoint for a specific record in UCSBOrganization table.
Add `DELETE` endpoint for a specific record in UCSBOrganization table.

# Acceptance Criteria:

- [ ] In `UCSBOrganizationController.java` there is code for an 
      endpoint `DELETE /api/UCSBOrganization?id=123` endpoint 
      that deletes the record if it exists, and returns 200 (ok) and 
      the text `record 123 deleted`, or returns 404 (Not Found) and
      the text `record 123 not found` if it does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBOrganizationController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBOrganizationController.java`

### LINKED ISSUE #23: Add `PUT` (edit) endpoint for a single record in UCSBOrganization table
Add `PUT` (edit) endpoint for a single record in UCSBOrganization table

Note that it is a good idea to implement `GET` for a single item
first; when using `PUT` manually, it's convenient to be able to copy/paste the JSON representation of the data and then edit it in the swagger interface.

Eventually, it's typically the case that in order to populate an edit
form in the front end, you first do a `GET` on a particular item, 
and use that JSON to populate the fields in a form.  You can send the
JSON representation of the new item to the backend.

# Acceptance Criteria:

- [ ] In `UCSBOrganizationController.java` there is code for an 
      endpoint `PUT /api/UCSBOrganization?id=123` endpoint 
      that accepts JSON for a new set of values for the database
      fields other than `id`, and updates the values of those fields.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBOrganizationController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBOrganizationController.java`

### LINKED ISSUE #24: Add `GET` (show) endpoint for a single record in UCSBOrganization table
Add `GET` (show) endpoint for a single record in UCSBOrganization table

# Acceptance Criteria:

- [ ] In `UCSBOrganizationController.java` there is code for an 
      endpoint `GET /api/UCSBOrganization?id=123` endpoint 
      that returns the JSON of the database record with id 123 if it
      exists, or a 400 and the error message `id 123 not found` if it
      does not.
- [ ] The Swagger-UI endpoints for this endpoint is well documented
      so that any member of the team can understand how to use it.
- [ ] The endpoint works as expected on localhost.
- [ ] The endpoint works as expected when deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the new code in 
      `UCSBOrganizationController.java`
- [ ] There is full mutation test coverage (Pitest) for new code in
      `UCSBOrganizationController.java`

### LINKED ISSUE #25: Create `UCSBOrganizationController`, add `GET` (index) and `POST` (create)
Create `UCSBOrganizationController`, add `GET` (index) and `POST` (create)

# Acceptance Criteria:

- [ ] There is a controller file `UCSBOrganizationController.java`
      in the expected directory.
- [ ] In `UCSBOrganizationController.java` there is 
      code for a `GET /api/UCSBOrganization/all` endpoint 
      that returns a JSON list of all `UCSBOrganization`s in the database.
      (We sometimes call this an *index* action since it lists all
      items in the database.)
- [ ] In `UCSBOrganizationController.java` there is 
      code for a `POST /api/UCSBOrganization/post` endpoint
      that can be used to create a new entry in the table. (This
      is a *create* action.)
- [ ] The Swagger-UI endpoints for these are well documented so that
      any member of the team can understand what they are for and
      how to use them.
- [ ] The `POST` endpoint works as expected, in the sense that new
      records can be added to the database (on localhost).
- [ ] The `GET` endpoint works as expected, in the sense that the new
      records that are added show up (on localhost).
- [ ] The `GET` and `POST` endpoints work as expected when the 
      app is deployed to Heroku.
- [ ] There is full test coverage (Jacoco) for the methods in 
      `UCSBOrganizationController.java`
- [ ] There is full mutation test coverage (Pitest) for the methods in
      `UCSBOrganizationController.java`

### LINKED ISSUE #26: Create database table for UCSBOrganization
Create database table for UCSBOrganization

# Acceptance Criteria:

- [ ] There is an `@Entity` class called UCSBOrganization.java
- [ ] There is a `@Repository` class called UCSBOrganizationRepository.java
- [ ] When you start up the repo on localhost, you can see the table
      using the H2 console (see the file `docs/h2-database.md` for 
      instructions.)
- [ ] When you deploy the app to Heroku, you can see the table
      using the psql command (See `Accessing Database Console` in the
      main `README.md` file for instructions.)

### PR DESCRIPTION
**Pull request by jwangathan**
Created Entity and Repository class for UCSB Organization. Created Controller and Tests for GET(index), POST(create), GET(show), PUT(edit), and DELETE for UCSB Organization. Closes #22 #23 #24 #25 #26

### CODE REVIEW

**Review by aaron-sgy — CHANGES_REQUESTED** [review]
# Test Issues

Needs better test. 2 mutations survived. 

Screen Shot 2022-05-01 at 14 07 36

Screen Shot 2022-05-01 at 14 07 24
.

# API

Api's GET/POST/DELETE/ all work great. However, for PUT, when you edit the orgCode in the incoming block, it will create a whole new entry. I think that might be normal? But not too sure. Need someone else to review this.

**Comment by yuheho7749** [comment]
The behavior of put/update generating a new entry is not intended.

The reason being, since the orgCode is the unique id (`@Id` annotation documentation can be found here <https://docs.oracle.com/javaee/7/api/javax/persistence/Id.html>), changing/setting the orgCode to something different in the controller and then calling the ucsbOrganizationRepository.save() function will create a new entry. Notice that the save function is used in both the post and put functions. In other words, if the Id is not found in the database table, it creates a new entry; if the Id is found, it modifies the entry

File: src/main/java/edu/ucsb/cs156/example/controllers/UCSBOrganizationController.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            @ApiParam("orgCode") @RequestParam String orgCode,
+            @RequestBody @Valid UCSBOrganization incoming) {
+
+        UCSBOrganization ucsbOrganization = ucsbOrganizationRepository.findById(orgCode)
+                .orElseThrow(() -> new EntityNotFoundException(UCSBOrganization.class, orgCode));
+
+        ucsbOrganization.setOrgCode(incoming.getOrgCode());
+        ucsbOrganization.setOrgTranslationShort(incoming.getOrgTranslationShort());
+        ucsbOrganization.setOrgTranslation(incoming.getOrgTranslation());
+        ucsbOrganization.setInactive(incoming.getInactive());
```
Comment (yuheho7749) [inline src/main/java/edu/ucsb/cs156/example/controllers/UCSBOrganizationController.java]: Needs to be changed to match the intended behavior of updating a database entry. Updating should not generate a new entry

**Review by yuheho7749 — CHANGES_REQUESTED** [review]
Needs to match intended behavior of put/update

**Review by yuheho7749 — APPROVED** [review]
LGTM

**Comment by aaron-sgy** [comment]
# Test Coverage

All tests passed. 100% mutation tests coverage.

# API: PUT Issue

Using "PUT": When updating an existing UCSBOrganization's orgCode, it no longer creates a new entry. The orgCode cannot be edited through the PUT request, but in the editing body, it still appears, making it seem like the orgCode is editable. I suggest deleting that as well. 

# TLDR

Can technically merge by ignoring the API issue since the API issue is not a critical issue that violates the intended API behavior. But solving that would improve the Webapp overall.

**Review by aaron-sgy — APPROVED** [review]
